restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -126,9 +129,23 @@ public void maybeCreateTable(String keyspaceName, String tableName) {
                   String.join(", ", allPathColumnNames)))
           .get();
 
-      dataStore
-          .query(String.format("CREATE INDEX ON \"%s\".\"%s\"(leaf)", keyspaceName, tableName))
-          .get();
+      // After creating the table, it can take up to 2 seconds for permissions cache to be updated,
+      // So retry this first index creation until it succeeds.  The other index creations should not
+      // need this behavior.
+      boolean success = false;
+      while (!success) {
```

Comment:
This had the potential to cause an infinite look if the index creation fails for an non-transient reason.

Reply:
This diff is actually inside a bigger `try` block - which exceptions would cause an infinite loop?

Reply:
either way though it probably makes sense to only retry a finite number of times

Reply:
> which exceptions would cause an infinite loop?

those that we catch on line 143 and ignore.

Reply:
true, duh - thanks